lib/ts/recipe/withOverride.tsx

```diff
@@ -0,0 +1,17 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import React, { useContext } from "react";
+import { ComponentOverrideContext } from "./componentOverrideContext";
+
+export const withOverride = <TComponent extends React.FunctionComponent<any>>(
+    overrideKey: string,
+    DefaultComponent: TComponent
+) => {
+    return (props: React.ComponentProps<TComponent>) => {
+        const OverrideComponent: TComponent = useContext(ComponentOverrideContext).get(overrideKey) as TComponent;
```

Comment:
- Yes, if the implementation changes to object, `.get()` method won't work.
- I'll try to do some type acrobatics to avoid it 😄

Reply:
> Yes, if the implementation changes to object, .get() method won't work.
The user is providing an object in the config. If you are converting that to a map, then you can continue to use .get. Otherwise, you should use `useContext(ComponentOverrideContext)[overrideKey]`

Reply:
> The user is providing an object in the config. If you are converting that to a map, then you can continue to use .get. Otherwise, you should use useContext(ComponentOverrideContext)[overrideKey]

I implemented it using an object, so it's solved.

> Can you avoid the `as TComponent` part?

It turns out I can fully omit it. The resulting type of a rendered component is `TComponent | React.FunctionComponent<any>`. We're getting a type from a string -> component factory map, so if we don't extract the mapping information from the type itself (e.g. specific `OverrideComponentMap`), we won't have the exact component type. 

This is not an issue though, because:
- with this implementation we get type hints for overriden component props (thanks to recipe-specific `ComponentOverrideMap`)
- compiler will not complain about it in `withOverride`, because `React.FunctionComponent<any>` is a broader type that can accept _any_ props.